src/software/estop/uart_communication.h

```diff
@@ -0,0 +1,72 @@
+#pragma once
+
+#include <boost/asio.hpp>
+#include <boost/function.hpp>
+
+/*
+ * A boost::asio wrapper class that is used to synchronously communicate with a serial
+ * device via UART
+ */
+
+class UartCommunication
+{
+    using io_service      = boost::asio::io_service;
+    using serial_port_ptr = std::shared_ptr<boost::asio::serial_port>;
+
+   public:
+    enum flush_type
+    {
+        flush_receive = TCIFLUSH,  // flushes data received but not read
+        flush_send    = TCOFLUSH,  // flushes data written but not transmitted
+        flush_both    = TCIOFLUSH  // flushes both data received but not read and data
+                                   // written but not transmitted
+    };
+
+    UartCommunication();
+
```

Comment:
Deleting the copy constructor would probably be a good idea here, if one of them closes the underlying serial port object then you would have issues trying to use it from the copied object.

Reply:
deleted